docs/spec_draft.md

```diff
@@ -1443,6 +1444,138 @@ specification. Numeric precision is implementation-defined.
+selected locations whose shape matches `source`. These are then scattered to the
+output which is initialized with `init_value`.  Multiple scattered elements
+which land in the same output location are combined using the `scatter`
+function.
+
+The following diagram shows how elements in `results` are computed from
+`operand` and `source` using a concrete example.
+
+<p align="center">
+  <img src="spec_draft/select_and_scatter.png" />
+</p>
+
+
+Formally,
+ * Apply [stablehlo.reduce_window](#stablehloreduce_window) with the following
+   parameters to get `selected_values`.
+   * `inputs` $=$ `{ operand }`,
+   * `init_values` = {identity value for `select`}.
+   * `window_dimensions`, `window_strides`, and `padding` are used as is.
+   * `body` $=$ `select`, to select an element from each window by applying it
+     across each window.
+ * Let `selected_indices = {index-of(value, operand: value in selected_values}`,
+   where `index-of(a, t)` provides the index of value `a` in tensor `t` as
+   selected by `select` function.
+ * `result[i0, i1, ..., iR'-1] = map({init_value} U {source[j0, j1, ..., jR-1]}, body)`
+   if
+   * `jd = floor(((padding[d] + id) + jd * (window_dimensions[d] - window_strides[d])) /  window_dimensions[d])`.
+   * `selected_indices[j0, j1, ..., jR-1] = (i0, i1, ..., iR'-1)`.
+ * `result[i0, i1, ..., iR'-1] = init_value`, otherwise.
+
+The evaluation order of the `scatter` function is arbitrary and may be
+non-deterministic. Therefore, the scatter function should not be overly
+sensitive to reassociation. See the discussion about associativity in the
+context of [stablehlo.reduce](#stablehloreduce) for more details.
+
+### Inputs
+
+| Name                | Type                                      | Constraints      |
+|---------------------|-------------------------------------------|------------------|
+| `operand`           | tensor of any supported type              | (C1-C10)         |
+| `source`            | tensor of any supported type              | (C1-C3)          |
+| `init_value`        | scalar of any supported type              | (C4), (C9)       |
+| `window_dimensions` | 1-dimensional tensor constant type `si64` | (C1), (C3), (C5) |
+| `window_strides`    | 1-dimensional tensor constant type `si64` | (C6)             |
+| `padding`           | 2-dimensional tensor constant type `si64` | (C3), (C7)       |
+| `select`            | `function`                                | (C8)             |
+| `scatter`           | `function`                                | (C4), (C9)       |
+
+### Outputs
+
+| Name     | Type                         | Constraints |
+|----------|------------------------------|-------------|
+| `result` | tensor of any supported type | (C9-C10)    |
+
+### Constraints
+
+  * (C1) rank(`source`) $=$ rank(`operand`) $=$ size(`window_dimensions`).
+
+  * (C2) `operand` and `source` have the same element type ignoring floating-point
```

Comment:
Do we need the "ignoring floating-point precision" part as part of the constraint?

Reply:
Thanks for the catch. No it is not needed.